582 - add release notes (#583)
eclipse/microprofile-config: spec/src/main/asciidoc/release_notes.asciidoc

// John D. Ament
// Emily Jiang

[[release_notes_20]]
== Release Notes for MicroProfile Config 2.0

A full list of changes delivered in the 2.0 release can be found at link:https://github.com/eclipse/microprofile-config/milestone/8?closed=1[MicroProfile Config 2.0 Milestone].

=== Incompatible Changes

- ConfigSource#getPropertyNames is no longer a default method. The implementation of a ConfigSource must implement this method. (link:https://github.com/eclipse/microprofile-config/issues/431[431])

=== API/SPI Changes

- Convenience methods have been added to Config allowing for the retrieval of multi-valued properties as lists instead of arrays (link:https://github.com/eclipse/microprofile-config/issues/496[#496])

> - Support Configuration Profile so that the corresponding properties associated with the active profile will be used (link:https://github.com/eclipse/microprofile-config/issues/418[#418])

Reviewer: ```suggestion
- Support Configuration Profiles so that the corresponding properties associated with the active profile are used (link:https://github.com/eclipse/microprofile-config/issues/418[#418])
```

Author: done

> - Enable isolation of config properties into a separate POJO by introducing `@ConfigProperties` (link:https://github.com/eclipse/microprofile-config/issues/240[240])

Reviewer 2: ```suggestion
- Enable bulk-extraction of config properties into a separate POJO by introducing `@ConfigProperties` (link:https://github.com/eclipse/microprofile-config/issues/240[240])
```

Author: done

- Enable users to determine the winning source for a configuration value (link:https://github.com/eclipse/microprofile-config/issues/312[312]) (link:https://github.com/eclipse/microprofile-config/issues/43[43])
- Expose conversion mechanism in Config API (link:https://github.com/eclipse/microprofile-config/issues/492[492])

=== Functional Changes
- Provide built-in Converters: OptinalInt, OptionalLong and OptionalDouble (link:https://github.com/eclipse/microprofile-config/issues/513[513])

> - Clarify behaviour of Converters for primitive wrappers (link:https://github.com/eclipse/microprofile-config/issues/520[520])

Reviewer 2: This wording requires the reader to chase and look for the effect of the clarification. I think it should state the outcome shortly, like 
```suggestion
- Clarifies that Converters for primitive wrappers apply to primitive types as well (link:https://github.com/eclipse/microprofile-config/issues/520[520])
```

Author: done
